src/Compilers/CSharp/Portable/Compilation/BuiltInOperators.cs

```diff
@@ -715,7 +715,7 @@ private TypeSymbol LiftedType(BinaryOperatorKind kind)
             return null;
         }
 
-        internal static bool IsValidObjectEquality(Conversions Conversions, TypeSymbol leftType, bool leftIsNullOrDefault, TypeSymbol rightType, bool rightIsNullOrDefault, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        internal static bool IsValidObjectEquality(Conversions Conversions, TypeSymbol leftType, bool leftIsNullOrDefault, bool leftIsNew, TypeSymbol rightType, bool rightIsNullOrDefault, bool rightIsNew, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
```

Comment:
>IsValidObjectEquality [](start = 29, length = 21)

:memo: I skipped this change in this review pass, as I assume it will be undone after adding support for `new(...)` in all binary operators.